climada/entity/exposures/crop_production.py

```diff
@@ -371,76 +367,86 @@ def set_to_usd(self, input_dir=INPUT_DIR, yearrange=YEARS_FAO):
+
+        Returns:
+            Exposure
         """
 
+        # the exposure in t/y is saved as 'value_tonnes'
+        self['value_tonnes'] = self['value']
+
         # account for the case of only specifying one year as yearrange
         if len(yearrange) == 1:
             yearrange = np.array([yearrange[0], yearrange[0]])
 
         # open both FAO files and extract needed variables
         # FAO_FILE: contains producer prices per crop, country and year
-        fao = pd.read_csv(os.path.join(input_dir, FAO_FILE))
-        fao_area = getattr(fao, 'Area Code').values
-        fao_crops = fao.Item.values
-        fao_year = fao.Year.values
-        fao_price = fao.Value.values
+        fao = dict()
+        fao['file'] = pd.read_csv(os.path.join(input_dir, FAO_FILE))
+        fao['crops'] = fao['file'].Item.values
+        fao['year'] = fao['file'].Year.values
+        fao['price'] = fao['file'].Value.values
 
-        fao_country = co.country_faocode2iso(fao_area)
+        fao_country = coord.country_faocode2iso(getattr(fao['file'], 'Area Code').values)
 
         # create a list of the countries contained in the exposure
         iso3alpha = list()
-        for item, _ in enumerate(self.region_id):
-            if (self.region_id[item] == 0) or (self.region_id[item] == -99):
-                iso3alpha.append('No country')
-            elif (self.region_id[item] == 902) or (self.region_id[item] == 910) or \
-            (self.region_id[item] == 914) or (self.region_id[item] == 915):
-                iso3alpha.append('Other country')
-            else:
-                iso3alpha.append(iso_cntry.get(self.region_id[item]).alpha3)
+        for reg_id in self.region_id:
+            try:
+                iso3alpha.append(iso_cntry.get(reg_id).alpha3)
+            except KeyError:
+                if reg_id in (0, -99):
+                    iso3alpha.append('No country')
+                else:
+                    iso3alpha.append('Other country')
         list_countries = np.unique(iso3alpha)
 
+
+
+
         # iterate over all countries that are covered in the exposure, extract the according price
-        # and calculate the produced yield in USD/y
+        # and calculate the crop production in USD/y
         area_price = np.zeros(self.value.size)
-        for item, _ in enumerate(list_countries):
-            country = list_countries[item]
-            if country != 'No country':
-                if country == 'Other country':
-                    price = 0
-                else:
-                    idx_price = np.where((np.asarray(fao_country) == country) &
-                                         (np.asarray(fao_crops) == (CROP_NAME[self.crop])['fao']) &
-                                         (fao_year >= yearrange[0]) & (fao_year <= yearrange[1]))
-                    price = np.mean(fao_price[idx_price])
+        for country in list_countries:
+            idx_country = np.where(np.asarray(iso3alpha) == country)[0]
```

Comment:
if the right-hand side is a list with a single element, you could also write
`[idx_country] = np.where(np.asarray(iso3alpha) == country)`